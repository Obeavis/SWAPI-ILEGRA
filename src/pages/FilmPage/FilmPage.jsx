import React, { useState, useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import { withRouter } from "react-router-dom";
import { Container, Row, Col } from "react-bootstrap";
import NotFound from "../NotFound/NotFound";
import SomethingWentWrong from "../SomethingWentWrong/SomethingWentWrong";
import BB8Loader from "../../components//BB8Loader/BB8Loader";
import StarWarsCrawlText from "../../components/StarWarsCrawlText/StarWarsCrawlText";


const FilmPage = (props) => {
	const [error, setError] = useState(false);
	const dispatch = useDispatch();

	useEffect(() => {
		const episode = parseInt(props.history.location.pathname.split('/')[2]);
		if (isNaN(episode)) {
			setError(true);
		} else {
			dispatch({
				type: "GET_FILM",
				filmID: episode
			});
		}
		// eslint-disable-next-line
	}, []);

	const film = useSelector(state => state.film);

	return (
		<div className="film-page" key={'uhasuhausha' + new Date()}>
			{film.error === true || Object.keys(film).length <= 0 ?
				<SomethingWentWrong />
				: error === true || film.filmNotFound === true ?
					<NotFound />
					: Object.keys(film.film).length <= 0 ?
						<Container>
							<Row className="justify-content-center mt-5">
								<BB8Loader />
							</Row>
						</Container>
						: <Container key={'StarWarsCrawlText'}>
							<Row>
								<Col className="StarWarsCrawlText" key={'teste'}>
									<StarWarsCrawlText film={film.film} key={'teste'}/>
								</Col>
							</Row>
						</Container>

			}
		</div>
	)
}

export default withRouter(FilmPage);