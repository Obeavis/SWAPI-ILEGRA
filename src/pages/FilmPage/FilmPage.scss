.film-page {
    .StarWarsCrawlText {
        max-height: 10px;
    }
}