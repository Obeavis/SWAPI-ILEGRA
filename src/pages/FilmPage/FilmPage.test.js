import React from "react";
import { Provider } from 'react-redux'
import { MemoryRouter } from 'react-router-dom';
import configureStore from "redux-mock-store";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import { configure, mount } from "enzyme";
import FilmPage from "./FilmPage";

configure({ adapter: new Adapter() });



describe("FilmPage", () => {
    const mockStore = configureStore();
    let store;

    it("Matches the snapshot", () => {
        store = mockStore({
            film: {
                "film": {
                    "episodeID": 1,
                    "title": "The Phantom Menace",
                    "director": "George Lucas",
                    "releaseDate": "1999-05-19",
                    "openingCrawl": "Turmoil has engulfed the\r\nGalactic Republic. The taxation\r\nof trade routes to outlying star\r\nsystems is in dispute.\r\n\r\nHoping to resolve the matter\r\nwith a blockade of deadly\r\nbattleships, the greedy Trade\r\nFederation has stopped all\r\nshipping to the small planet\r\nof Naboo.\r\n\r\nWhile the Congress of the\r\nRepublic endlessly debates\r\nthis alarming chain of events,\r\nthe Supreme Chancellor has\r\nsecretly dispatched two Jedi\r\nKnights, the guardians of\r\npeace and justice in the\r\ngalaxy, to settle the conflict...."
                },
                "error": false,
                "filmNotFound": false
            }
        });

        const wrapper = mount(
            <Provider store={store}>
                <MemoryRouter>
                    <FilmPage />
                </MemoryRouter>
            </Provider>
        );

        const tree = renderer.create(wrapper).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it("Renders the <FilmPage />", () => {
        const wrapper = mount(
            <Provider store={store}>
                <MemoryRouter initialEntries={['/films/1']}>
                    <FilmPage />
                </MemoryRouter>
            </Provider>
        );

        expect(wrapper.find('FilmPage').exists()).toEqual(true)
    });

    it("Renders <BB8Loader />", () => {
        store = mockStore({
            film: {
                "film": {},
                "error": false,
                "filmNotFound": false
            }
        });

        const wrapper = mount(
            <Provider store={store}>
                <MemoryRouter initialEntries={['/films/1']}>
                    <FilmPage />
                </MemoryRouter>
            </Provider>
        );

        expect(wrapper.find('BB8Loader').exists()).toEqual(true)
    });

    it("Renders <SomethingWentWrong />", () => {
        store = mockStore({
            film: {}
        });

        const wrapper = mount(
            <Provider store={store}>
                <MemoryRouter>
                    <FilmPage initialEntries={['/films/1']} />
                </MemoryRouter>
            </Provider>
        );

        expect(wrapper.find('SomethingWentWrong').exists()).toEqual(true)
    });

})