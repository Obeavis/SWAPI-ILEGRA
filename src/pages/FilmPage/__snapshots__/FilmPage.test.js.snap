// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmPage Matches the snapshot 1`] = `
<div
  className="film-page"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="StarWarsCrawlText col"
      >
        <div
          className="star-wars-intro"
        >
          <p
            className="intro-text"
          >
            A long time ago in a galaxy far, 
            <br />
            far away...
          </p>
          <h2
            className="main-logo"
          >
            <img
              alt="The Phantom Menace"
              src="star-wars-logo.png"
            />
          </h2>
          <div
            className="main-content"
          >
            <div
              className="title-content"
            >
              <p
                className="content-header"
              >
                EPISODE 1
                 
                <br />
                The Phantom Menace
              </p>
              <p
                className="content-body"
              >
                Turmoil has engulfed the
Galactic Republic. The taxation
of trade routes to outlying star
systems is in dispute.

Hoping to resolve the matter
with a blockade of deadly
battleships, the greedy Trade
Federation has stopped all
shipping to the small planet
of Naboo.

While the Congress of the
Republic endlessly debates
this alarming chain of events,
the Supreme Chancellor has
secretly dispatched two Jedi
Knights, the guardians of
peace and justice in the
galaxy, to settle the conflict....
                <br />
                <br />
                <span>
                  Direct by: 
                  George Lucas
                </span>
                <br />
                <br />
                <span>
                  Release Date: 
                  <time
                    dateTime={"1999-05-19T00:00:00.000Z"}
                  >
                    May 18, 1999
                  </time>
                </span>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
